src/base.jl

```diff
@@ -0,0 +1,24 @@
+# this file contains code copied from Base
+# multidimensional.jl, 289
```

Comment:
Can we replace this with a hyperlink to the commit

Reply:
https://github.com/JuliaLang/julia/blob/f12cde0e8c7f4268838d3201dd9e13b90ea44633/base/multidimensional.jl#L761-L777

Reply:
but also, can this be deleted? do we only actually use the `_unsafe_getindex!` defined below?

Reply:
i have commented it out for now... we may or may not want it (needs benchmarks i suppose)